core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -966,8 +965,9 @@ private void setRoot(InodeDirectory directory) {
    * appropriate inode indexes.
    *
    * @param inode the inode to add to the inode tree
+   * @param hasMode if the journal entry has mode set
    */
-  private void addInodeFromJournalInternal(Inode<?> inode) {
+  private void addInodeFromJournalInternal(Inode<?> inode, boolean hasMode) {
```

Comment:
would it work to use inode.getMode() instead of adding another parameter?

Reply:
let me check what inodeentry.getMode returns if there was no mode set in the protobuf entry. if we can identify this as invalid mode, we might be fine.

Reply:
@aaudiber the journal entry mode will return 0 if it is not set which is a valid mode

Reply:
can we check `inode.hasMode()` instead?